src/main/java/ch/njol/skript/effects/EffBan.java

```diff
@@ -73,15 +76,17 @@
 	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
 		players = exprs[0];
 		reason = exprs.length > 1 ? (Expression<String>) exprs[1] : null;
+		expires = exprs.length > 1 ? (Expression<Timespan>) exprs[2] : null;
 		ban = matchedPattern % 2 == 0;
 		ipBan = matchedPattern >= 2;
 		return true;
 	}
 	
+	@SuppressWarnings("null")
 	@Override
 	protected void execute(final Event e) {
 		final String reason = this.reason != null ? this.reason.getSingle(e) : null; // don't check for null, just ignore an invalid reason
-		final Date expires = null;
+		final Date expires = this.expires != null ? new Date(System.currentTimeMillis() + this.expires.getSingle(e).getMilliSeconds()) : null;

```

Comment:
getSingle may be null, so it's not safe to call getMS right away

Reply:
Arg Skript ... I tell ya! haha! I was trying to keep this as neat and tidy as possible, but nope. Ill do a quick little fix for this, hope it doesn't get too messy.